crates/sui-types/src/crypto.rs

```diff
@@ -619,18 +753,40 @@ pub fn sha3_hash<S: Signable<Sha3_256>>(signable: &S) -> [u8; 32] {
     hash.into()
 }
 
+// 
+// Helper enum to statically dispatch sender sigs to verification obligation.
```

Comment:
Helper enums that matches signature if it is authority signature.

Reply:
The verification obligation will convert all signatures into this enum form before processing it.